pure_protobuf/dataclasses_.py

```diff
@@ -117,6 +108,21 @@ def optional_field(number: int, *args, **kwargs) -> Any:
     return field(number, *args, default=None, **kwargs)
 
 
+def one_of(**parts: Tuple[type, int]) -> OneOf_:
+    scheme_ = tuple(
+        OneOfPartInfo(name, type_, number)
+        for name, (type_, number) in parts.items()
+    )
+
+    default = OneOf_(scheme_)
+    return _field(-1, isoneof=True, default=default)
```

Comment:
Does the `-1` get handled somewhere? I must be overlooking it

Reply:
Kinda yes. The trick here is that from `OneOf_` there will be created fields in   `__protobuf_fields__` only for its parts, and `OneOf_` is never a part of a message, it's only sugar.

For example for `A`:
```python
@message
@dataclass
class A:
    msg: OneOf_ = one_of(
        a=part(1, int32),
		b=part(2, str)
    )
```

`__protobuf_fields__` gonna look like that:

```python
__protobuf_fields__ = {
    1: OneOfPartField(1, name="msg", origin=Int32Field(name='a', ...), ...),
    2: OneOfPartField(2, name="msg", origin=StrField(name='b', ...), ...),
}
```
there won't be any field for `OneOf_`i.e. for `msg` itself, only for it's parts

Reply:
Hmm, yeah I see. I don't really like passing around a fake value which is never used. It suggests that we should split `_field` function. Maybe later (tm)